src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -268,13 +267,18 @@ public function get_recipients( $thread_id = 0 ) {
 
 		$recipients = wp_cache_get( 'thread_recipients_' . $thread_id, 'bp_messages' );
 		if ( false === $recipients ) {
-			$bp = buddypress();
 
 			$recipients = array();
-			$sql        = $wpdb->prepare( "SELECT * FROM {$bp->messages->table_name_recipients} WHERE thread_id = %d", $thread_id );
-			$results    = $wpdb->get_results( $sql );
+			/*$sql        = $wpdb->prepare( "SELECT * FROM {$bp->messages->table_name_recipients} WHERE thread_id = %d", $thread_id );
+			$results    = $wpdb->get_results( $sql );*/
+			$results = BP_Messages_Thread::get(
+				array(
+					'per_page'        => - 1,
```

Comment:
@janakkaneriya007 Here too. we should not pass `-1` as `per_page`

Reply:
@dipskakadiya old query didn't have any pagination values in it so we can't limit the results to 20 which is by default.